plugin/src/main/java/org/opensearch/ml/action/controller/CreateControllerTransportAction.java

```diff
@@ -98,6 +103,9 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<MLCrea
         String[] excludes = new String[] { MLModel.MODEL_CONTENT_FIELD, MLModel.OLD_MODEL_CONTENT_FIELD };
 
         try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            if (!mlFeatureEnabledSetting.isControllerEnabled()) {
```

Comment:
I just realized why do we need to throw the exception after stashing the context? We can throw this exception before stashing the context, right?

Reply:
Yes you are right. I just followed other's patterns.

Reply:
Just verified with UT that we cannot do this movement. This is because if we move the exception outside it will throw the exception directly, instead of catching by `actionlistener`.

Reply:
What's the difference? how stashing context help to catch by `actionlistener`?

Reply:
Actually I am not so sure about why, but UT's behaved in such way

Reply:
Maybe we can see if @ylwu-amzn has any output here?

Reply:
Because in the catch block we perform `actionListener.onFailure(e);` to catch the exception. Let's keep the code as it is then.